File: demo-gigya-java/src/main/java/com/ownid/demo/gigya/ui/activity/UserActivity.java
package com.ownid.demo.gigya.ui.activity;

import android.content.Intent;

import com.gigya.android.sdk.Gigya;
import com.gigya.android.sdk.GigyaCallback;
import com.gigya.android.sdk.account.models.GigyaAccount;
import com.gigya.android.sdk.account.models.Profile;
import com.gigya.android.sdk.network.GigyaError;
import com.ownid.demo.ui.activity.BaseUserActivity;

public class UserActivity extends BaseUserActivity {

    @Override
    public void signOut() {
        Gigya.getInstance(GigyaAccount.class).logout();
    }

    @Override
    public void startMainActivity() {
        startActivity(new Intent(UserActivity.this, MainActivity.class));
    }

    @Override
    protected void onResume() {
        super.onResume();

        Gigya.getInstance(GigyaAccount.class)
                .getAccount(true, new GigyaCallback<GigyaAccount>() {
                    @Override
                    public void onSuccess(GigyaAccount account) {
                        if (account == null) {
                            startMainActivity();
                            finish();
                        } else {
                            final Profile profile = account.getProfile();
                            if (profile != null) {
                                String displayName = profile.getFirstName();
                                if (displayName == null) displayName = "";

                                String email = profile.getEmail();
                                if (email == null) email = "";
                                showUser(displayName, email);
                            }
                        }
                    }

                    @Override
                    public void onError(GigyaError error) {
                        if (error != null) showError(error.getLocalizedMessage());
                        else showError("Unknown error");
                    }
                });
    }
}

File: demo-gigya-screens/src/main/java/com/ownid/demo/gigya/screens/ui/UserActivity.kt
package com.ownid.demo.gigya.screens.ui

import android.content.Intent
import com.gigya.android.sdk.Gigya
import com.gigya.android.sdk.GigyaCallback
import com.gigya.android.sdk.account.models.GigyaAccount
import com.gigya.android.sdk.network.GigyaError
import com.ownid.demo.ui.activity.BaseUserActivity

class UserActivity : BaseUserActivity() {

    private val gigya by lazy(LazyThreadSafetyMode.NONE) { Gigya.getInstance(GigyaAccount::class.java) }

    override fun signOut() = gigya.logout()
    override fun startMainActivity() = startActivity(Intent(this, MainActivity::class.java))

    override fun onResume() {
        super.onResume()

        gigya.getAccount(true, object : GigyaCallback<GigyaAccount>() {
            override fun onSuccess(account: GigyaAccount?) {
                if (account == null) {
                    startMainActivity()
                    finish()
                } else {
                    showUser(account.profile?.firstName ?: "", account.profile?.email ?: "")
                }
            }

            override fun onError(error: GigyaError) = showError(Throwable(error.toString()))
        })
    }
}